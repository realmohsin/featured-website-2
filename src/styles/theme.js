import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
