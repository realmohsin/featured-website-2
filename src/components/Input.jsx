import React from 'react'
import clsx from 'clsx'
import { makeStyles, useTheme } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  inputContainer: {
    position: 'relative',
    margin: '2rem 0'
  },
  input: {
    color: theme.palette.primary.dark,
    fontSize: '1.6rem',
    padding: '1rem 1rem 1rem 0.5rem',
    width: '100%',
    border: 'none',
    borderBottom: `1px solid ${theme.palette.secondary.main}`,
    background: 'transparent',
    margin: '2.5rem 0',
    '&:focus': {
      outline: 'none'
    },
    '&:focus ~ label': {
      top: '-1.4rem',
      fontSize: '1.1rem',
      color: theme.palette.secondary.main
    }
  },
  label: {
    color: '#757575',
    color: theme.palette.secondary.main,
    position: 'absolute',
    left: '0.5rem',
    top: '1.4rem',
    fontSize: '1.6rem',
    textTransform: 'uppercase',
    letterSpacing: '1px',
    transition: '300ms ease all',
    pointerEvents: 'none',
    [theme.breakpoints.down('xs')]: {
      fontSize: '1.4rem'
    }
  },
  shrunkLabel: {
    top: '-1.4rem',
    fontSize: '1.1rem',
    color: theme.palette.secondary.main
  },
  textarea: {
    border: `1px solid ${theme.palette.secondary.main}`,
    marginTop: '5rem',
    padding: '1rem'
  },
  textareaLabel: {
    left: '1.5rem'
  }
}))

const Input = ({ isTextarea, label, handleChange, ...inputProps }) => {
  const theme = useTheme()
  const classes = useStyles()

  return (
    <div className={classes.inputContainer}>
      {isTextarea ? (
        <textarea
          className={clsx(classes.input, classes.textarea)}
          onChange={handleChange}
          {...inputProps}
        />
      ) : (
        <input
          className={classes.input}
          onChange={handleChange}
          {...inputProps}
        />
      )}

      {label && (
        <label
          className={clsx(
            classes.label,
            inputProps.value.length > 0 && classes.shrunkLabel,
            isTextarea && classes.textareaLabel
          )}
        >
          {label}
        </label>
      )}
    </div>
  )
}

export default Input
