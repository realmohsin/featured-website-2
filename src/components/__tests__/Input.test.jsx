import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core/styles";
import { createMuiTheme } from "@material-ui/core/styles";
import Input from "../Input";

const theme = createMuiTheme({
  palette: {
    primary: { main: "#0E5C60", dark: "#0A4A4D" },
    secondary: { main: "#CCAB63" },
  },
  breakpoints: {
    down: () => "@media (max-width: 600px)",
  },
});

const renderWithTheme = (component) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};

describe("Input Component", () => {
  it("renders regular input field", () => {
    const { container } = renderWithTheme(
      <Input
        type="text"
        name="firstName"
        value=""
        label="First Name"
        onChange={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders textarea when isTextarea is true", () => {
    const { container } = renderWithTheme(
      <Input
        isTextarea
        name="message"
        value=""
        label="Message"
        rows={5}
        onChange={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with filled value (shrunk label)", () => {
    const { container } = renderWithTheme(
      <Input
        type="email"
        name="email"
        value="test@example.com"
        label="Email"
        onChange={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders without label", () => {
    const { container } = renderWithTheme(
      <Input
        type="text"
        name="noLabel"
        value=""
        onChange={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls onChange when input value changes", () => {
    const mockOnChange = jest.fn();
    renderWithTheme(
      <Input
        type="text"
        name="testInput"
        value=""
        label="Test Input"
        onChange={mockOnChange}
      />
    );

    const input = screen.getByRole("textbox");
    fireEvent.change(input, { target: { value: "new value" } });
    
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });

  it("calls onChange when textarea value changes", () => {
    const mockOnChange = jest.fn();
    renderWithTheme(
      <Input
        isTextarea
        name="testTextarea"
        value=""
        label="Test Textarea"
        rows={3}
        onChange={mockOnChange}
      />
    );

    const textarea = screen.getByRole("textbox");
    fireEvent.change(textarea, { target: { value: "new textarea value" } });
    
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });

  it("applies correct props to input element", () => {
    renderWithTheme(
      <Input
        type="email"
        name="email"
        value="test@example.com"
        label="Email Address"
        placeholder="Enter your email"
        required
        onChange={() => {}}
      />
    );

    const input = screen.getByRole("textbox");
    expect(input).toHaveAttribute("type", "email");
    expect(input).toHaveAttribute("name", "email");
    expect(input).toHaveAttribute("placeholder", "Enter your email");
    expect(input).toHaveAttribute("required");
    expect(input).toHaveValue("test@example.com");
  });

  it("applies correct props to textarea element", () => {
    renderWithTheme(
      <Input
        isTextarea
        name="message"
        value="Test message"
        label="Message"
        rows={5}
        cols={50}
        placeholder="Enter your message"
        onChange={() => {}}
      />
    );

    const textarea = screen.getByRole("textbox");
    expect(textarea.tagName).toBe("TEXTAREA");
    expect(textarea).toHaveAttribute("name", "message");
    expect(textarea).toHaveAttribute("rows", "5");
    expect(textarea).toHaveAttribute("cols", "50");
    expect(textarea).toHaveAttribute("placeholder", "Enter your message");
    expect(textarea).toHaveValue("Test message");
  });
});
