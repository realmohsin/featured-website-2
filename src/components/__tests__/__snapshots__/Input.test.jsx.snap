// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders regular input field 1`] = `
<div
  class="makeStyles-inputContainer-1"
>
  <input
    class="makeStyles-input-2"
    name="firstName"
    type="text"
    value=""
  />
  <label
    class="makeStyles-label-3"
  >
    First Name
  </label>
</div>
`;

exports[`Input Component renders textarea when isTextarea is true 1`] = `
<div
  class="makeStyles-inputContainer-7"
>
  <textarea
    class="makeStyles-input-8 makeStyles-textarea-11"
    name="message"
    rows="5"
  />
  <label
    class="makeStyles-label-9 makeStyles-textareaLabel-12"
  >
    Message
  </label>
</div>
`;

exports[`Input Component renders with filled value (shrunk label) 1`] = `
<div
  class="makeStyles-inputContainer-13"
>
  <input
    class="makeStyles-input-14"
    name="email"
    type="email"
    value="test@example.com"
  />
  <label
    class="makeStyles-label-15 makeStyles-shrunkLabel-16"
  >
    Email
  </label>
</div>
`;

exports[`Input Component renders without label 1`] = `
<div
  class="makeStyles-inputContainer-19"
>
  <input
    class="makeStyles-input-20"
    name="noLabel"
    type="text"
    value=""
  />
</div>
`;
